### PR TITLE
Option for ObstacleLayer to not override StaticLayer's unknown parts

### PR DESCRIPTION
**Pull request by RBT22**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | [#3603](https://github.com/ros-planning/navigation2/issues/3603) |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | gazebo simulation |

---

## Description of contribution in a few bullet points

* I added an updateWithMaxWithoutUnknownOverride method to the CostmapLayer
* I added an value option to the ObstacleLayer's  combination_method parameter that uses the previous method.

## Description of documentation updates required from your changes

* Added new parameter value, so need to add that to the documentation page
* 

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

**Review by AlexeyMerzlyakov — COMMENTED** [review]
In general, good idea and approach.
Please follow the review items from below, and also add the testcase(s) covering new functionality. + Documentation updates on https://navigation.ros.org/configuration/packages/costmap-plugins/obstacle.html parameters page are necessary for this change.

File: nav2_costmap_2d/plugins/obstacle_layer.cpp [hunk 1]
```diff
@@ -546,6 +546,9 @@ ObstacleLayer::updateCosts(
     case 1:  // Maximum
       updateWithMax(master_grid, min_i, min_j, max_i, max_j);
       break;
+    case 2: // Maximum without overwrite if unknown
```
[the remark this answers is not in the document]
Comment (RBT22) [reply nav2_costmap_2d/plugins/obstacle_layer.cpp]: I made some updates. Is this a good way to handle the conversions?

**Comment by SteveMacenski** [comment]
Your PR has some linting issues, make sure to run ament_uncrustify and ament_cpplint https://app.circleci.com/pipelines/github/ros-planning/navigation2/9493/workflows/a75513af-b769-408c-ad12-b576c2a60575/jobs/30458/tests

File: nav2_costmap_2d/plugins/obstacle_layer.cpp [hunk 2]
```diff
@@ -760,4 +768,22 @@ ObstacleLayer::resetBuffersLastUpdated()
   }
 }
 
+CombinationMethod combination_method_from_int (const int value, const std::string function_name){
```
Comment (SteveMacenski) [inline nav2_costmap_2d/plugins/obstacle_layer.cpp:771]: Remove function name, that's a pretty odd choice. Also, use the node's logger, don't create your own

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

**Comment by RBT22** [comment]
Thanks for the feedback! I tried to fix the problems you wrote.
Could you help me what kind of test is required for this functionality?

**Comment by SteveMacenski** [comment]
Looks good to me now. I think there's a need for documentation on the new enum type in the configuration guides as Alexey pointed out. 

>  Could you help me what kind of test is required for this functionality?

First, I'd just test locally on your robots to make sure it works in general. Then, add a new test to the `integration/obstacle_tests.cpp` file for setting the different combination method with a costmap with unknown background costs and making sure even if you insert information that its never updated on `updateCosts`. this probably involves making a new test fixture, so you could also add a new file to `unit` tests to do this as well, but that other file might be a good reference for you!

**Comment by RBT22** [comment]
I'm not sure if I understood correctly what you meant, but I uploaded an integration test. @SteveMacenski 
Also started the docs. :)

File: nav2_costmap_2d/test/integration/obstacle_tests.cpp [hunk 3]
```diff
@@ -554,3 +554,50 @@ TEST_F(TestNode, testDynParamsSetStatic)
...
+  ~TestNodeWithoutUnknownOverwrite() {}
+
+protected:
+  std::shared_ptr<TestLifecycleNode> node_;
+};
+
+/**
+ * Test CombinationMethod::MaxWithoutUnknownOverwrite in ObstacleLayer.
+ */
+TEST_F(TestNodeWithoutUnknownOverwrite, testMaxWithoutUnknownOverwriteCombinationMethod) {
```
Comment (SteveMacenski) [inline nav2_costmap_2d/test/integration/obstacle_tests.cpp:583]: Can you show that these same inputs with MaxOverwrite does change the unknown count?

**Comment by SteveMacenski** [comment]
Just that and otherwise looks good to me! It would be good to test the counter-case

**Review by SteveMacenski — APPROVED** [review]

**Pull request merged**